chainstate/tx-verifier/src/transaction_verifier/timelock_check.rs

```diff
@@ -133,5 +153,60 @@ where
         }
     }
 
+    let outputs = match tx.outputs() {
+        Some(outputs) => outputs,
+        None => return Ok(()),
+    };
+
+    if let Some(output_check_required) = output_check_required {
+        let block_height = tx_source.expected_block_height();
+        for output in outputs {
+            match output {
+                TxOutput::Transfer(_, _)
+                | TxOutput::Burn(_)
+                | TxOutput::CreateStakePool(_, _)
+                | TxOutput::ProduceBlockFromStake(_, _)
+                | TxOutput::CreateDelegationId(_, _)
+                | TxOutput::DelegateStaking(_, _) => Ok(()),
+                TxOutput::LockThenTransfer(_, _, timelock) => match output_check_required {
+                    OutputTimelockCheckRequired::DelegationSpendMaturity => {
+                        let required =
+                            chain_config.as_ref().spend_share_maturity_distance(block_height);
+                        check_output_timelock(timelock, required)
+                    }
+                    OutputTimelockCheckRequired::DecommissioningMaturity => {
+                        let required =
+                            chain_config.as_ref().decommission_pool_maturity_distance(block_height);
+                        check_output_timelock(timelock, required)
+                    }
+                },
+            }?;
+        }
+    }
+
     Ok(())
 }
+
+fn check_output_timelock(
```

Comment:
This function is kind of exactly done in `ChainstateRef::check_block_reward_timelock()`. Can we somehow unify them?